core/src/main/java/com/amplifyframework/core/DeveloperMenuActivity.java

```diff
@@ -16,16 +16,112 @@
+    // be shaken in order to make the developer menu appear or disappear.
+    private static final int SHAKE_TIME = 500;
+
+    // Listen to accelerometer sensor events.
+    private final SensorEventListener sensorEventListener = new SensorEventListener() {
+        @Override
+        public void onSensorChanged(SensorEvent sensorEvent) {
+            float xAccel = sensorEvent.values[0]; // acceleration in the x-direction
+            float yAccel = sensorEvent.values[1]; // acceleration in the y-direction
+            float zAccel = sensorEvent.values[2]; // acceleration in the z-direction
+            double curAcceleration = Math.sqrt(((xAccel * xAccel) + (yAccel * yAccel) + (zAccel * zAccel)));
+            if (curAcceleration > SHAKE_THRESHOLD) {
+                long currentTime = Time.now();
+                if (shakeStart == 0) {
+                    shakeStart = currentTime;
+                } else if (currentTime - shakeStart > SHAKE_TIME) {
+                    changeVisibility();
+                    shakeStart = 0;
+                }
+            }
+        }
+
+        @Override
+        public void onAccuracyChanged(Sensor sensor, int accuracy) { }
+    };
+
+    // Manager for the device's sensors.
+    private SensorManager sensorManager;
+    // The accelerometer sensor associated with the device.
+    private Sensor accelerometer;
+    // The time (in milliseconds) that the device started shaking
+    // (or 0 if the device is not shaking).
+    private long shakeStart;
+    // The parent layout for the developer menu.
+    private View devMenuLayout;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_dev_menu);
+        sensorManager = (SensorManager) getSystemService(Context.SENSOR_SERVICE);
+        if (sensorManager != null) {
+            accelerometer = sensorManager.getDefaultSensor(Sensor.TYPE_ACCELEROMETER);
+        }
+        devMenuLayout = findViewById(R.id.devLayout);
+        devMenuLayout.setFocusable(true);
+        devMenuLayout.setVisibility(View.GONE);
+        shakeStart = 0;
+    }
+
+    @Override
+    protected void onResume() {
+        if (accelerometer != null) {
+            sensorManager.registerListener(sensorEventListener, accelerometer, SensorManager.SENSOR_DELAY_NORMAL);
+        }
+        super.onResume();
+    }
+
+    @Override
+    protected void onPause() {
+        if (accelerometer != null) {
+            sensorManager.unregisterListener(sensorEventListener);
+        }
+        super.onPause();
+    }
+
+    @Override
+    public boolean onKeyUp(int keyCode, KeyEvent event) {
+        // KEYCODE_MENU is the code for pressing ctrl
+        // (or command) + m
+        if (keyCode == KeyEvent.KEYCODE_MENU) {
+            changeVisibility();
+            return true;
+        } else {
+            return super.onKeyUp(keyCode, event);
```

Comment:
Should super.onKeyUp always be called?

Reply:
In examples in the Android documentation, they override onKeyUp in a similar manner (only returning super.onKeyUp if the keycode doesn't match a certain value). Also, if I change the code to always return super.onKeyUp, even if the keycode matches the one for ctrl+m, then the visibility of the developer menu is only changed the first two times I press ctrl+m.